### ISSUE TITLE
AVT 3 - React Select issues with VoiceOver on iOS 

### ISSUE THREAD

**Issue by aledavila** [carbon-design-system/carbon#4452]
## Environment

tested in iOS 13.1.3
using VoiceOver

## Detailed description

Note: The menu focus is not consistent. The first couple of times it opened it was on "Choose an Option" and that option was checked (see screenshot). On reload and attempting it again there was no focus at all.

![image](https://user-images.githubusercontent.com/12533409/67505350-b2420b00-f650-11e9-8393-db5cbb60f29a.png)

## Steps to reproduce the issue

1. Swipe right, VO announces "Select main landmark"
2. Swipe right, VO announces "Optional helper text"
3. Swipe right to navigate to the select box. VoiceOver announces "Select. Choose an option pop-up button, double tap to activate the picker."
4. Double tap to open the menu, VoiceOver does not announce anything.

Expected Result: VO should announce that it is on a menu and the label of the option that has focus.

ref: #3338

**Comment by dakahn** [comment]
So where we are is:
- native select doesn't have a placeholder attribute. The go to solution for this is to slap a bunch of attributes on an `<option>` so it functions as the default selection, but is not accessible
- the current implementation of Select w/categories -- the categories are not accessible in JAWS on Windows 10 and are janky in VO for macOS/iOS

**Comment by aledavila** [comment]
As DA mentioned leaving it as a native select has the above issues with it. One possible suggestion is to make this component a combobox instead example (https://dequelabs.github.io/combobo/demo/). In doing so we could make it accessible and more customizable. Could be a variant of combobox since we already have our own code for it. 

Some implications to consider are:
 - How moving away from native select will affect it being a form item?
- Are there issues with no longer using native select? 
- Will moving away add unnecessary bulk to the component? 
- How would it be styled?

**Comment by joshblack** [comment]
Is this tied specifically to Select with categories? Or select in general?

**Comment by asudoh** [comment]
Running `<Select>` of latest `master` with iOS 13.2.3, VoiceOver announces the following when user double-taps on the select box:

"Select"
"Choose an option"
"Popup button"
"Choose an option"
"Selected"
"Picker item"
"Adjustable"
"1 of 7"
